orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/job/WaitOnJobCompletion.groovy

```diff
@@ -52,13 +53,26 @@ public class WaitOnJobCompletion extends AbstractCloudProviderAwareTask implemen
   JobUtils jobUtils
 
   static final String REFRESH_TYPE = "Job"
+  /**
+   * Extra minute to pad the timing supplied by the job provider.
+   * E.g. if TitusJobRunner says this task is limitted to 50minutes we will wait 50m + 5m(padding),
+   * we should wait a bit longer to allow for any inaccuracies of the clock across the systems
+   */
+  static final Long PROVIDER_PADDING_MS = Duration.ofMinutes(5).toMillis()
 
   @Override
   long getDynamicTimeout(@Nonnull StageExecution stage) {
+    Long jobTimeoutFromProvider = (stage.context.get("jobRuntimeLimitMs") as Long)
+
```

Comment:
Oh the hinted runout takes precedence.

Reply:
yeah, i was debating taking the max, but at the end, since this is to reduce confusion, i think job timeout makes sense to be the "main" param since the users know about it but not the stageoverride